### ISSUE TITLE
Handling mouse events on status bar

### ISSUE THREAD

**Issue by Arc676** [sminez/penrose#241]
TL;DR goal: Reproduce i3 functionality where you can click on a workspace in the status bar to focus it.

Is it possible to handle a mouse event in the status bar, or access the status bar from the `WindowManager`? Currently, mouse event handlers only take the window manager as an argument, so I'm not sure how to handle the mouse event based on its position relative to elements in the status bar.

It would be nice if one could implement a hook such that clicking on a workspace in the status bar would focus that workspace. I also like how scrolling on the workspaces has the same behavior as `WindowManager::cycle_workspace`, but this last feature is only tangentially related to the question because it can be implemented solely based on the mouse event's position. I realize one could do this for the workspaces as well if its size were fixed, but see #240.

**Comment by sminez** [comment]
Mouse event handling is extremely minimal at present and unlikely to get much attention any time soon (if at all) I'm afraid. The status bar provided from Penrose itself is deliberately minimal so you are unlikely to ever see feature parity with any other status bar you might have used before :slightly_smiling:
